core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -719,8 +720,8 @@ class DynamicListenerConfig(server: KafkaServer) extends BrokerReconfigurable wi
     val oldListeners = listenersToMap(oldConfig.listeners)
     if (!newAdvertisedListeners.keySet.subsetOf(newListeners.keySet))
       throw new ConfigException(s"Advertised listeners '$newAdvertisedListeners' must be a subset of listeners '$newListeners'")
-    if (newListeners.keySet != newConfig.listenerSecurityProtocolMap.keySet)
-      throw new ConfigException(s"Listeners '$newListeners' and listener map '${newConfig.listenerSecurityProtocolMap}' don't match")
+    if (!newListeners.keySet.subsetOf(newConfig.listenerSecurityProtocolMap.keySet))
```

Comment:
In current code, we are not allowing to add a new listener with default listenerSecurityProtocolMap. 

For example, I have a broker with below configs
listener.security.protocol.map = PLAINTEXT:PLAINTEXT,SSL:SSL,SASL_PLAINTEXT:SASL_PLAINTEXT,SASL_SSL:SASL_SSL
listeners=SASL_PLAINTEXT://:9092

Now I want to add  "PLAINTEXT://:9093" listeners config.  listener addition will fail at above check.

Reply:
@omkreddy Thanks for the PR, the change looks good.